widget/embedded/src/components/CustomDestination/CustomDestination.tsx

```diff
@@ -0,0 +1,170 @@
+import type { PropTypes } from './CustomDestination.types';
+import type { MouseEvent } from 'react';
+
+import { i18n } from '@lingui/core';
+import {
+  Alert,
+  ChevronDownIcon,
+  CloseIcon,
+  Divider,
+  IconButton,
+  PasteIcon,
+  Typography,
+  WalletIcon,
+} from '@rango-dev/ui';
+import React, { useEffect, useRef, useState } from 'react';
+
+import { useAppStore } from '../../store/AppStore';
+import { useQuoteStore } from '../../store/quote';
+import {
+  getBlockchainDisplayNameFor,
+  getBlockchainShortNameFor,
+} from '../../utils/meta';
+import { isValidAddress } from '../ConfirmWalletsModal/ConfirmWallets.helpers';
+import { CustomCollapsible } from '../CustomCollapsible/CustomCollapsible';
+import { ExpandedIcon } from '../CustomCollapsible/CustomCollapsible.styles';
+
+import {
+  alarmsStyles,
+  Container,
+  CustomDestinationButton,
+  StyledTextField,
+} from './CustomDestination.styles';
+
+const OPEN_DELAY = 500;
+
+export function CustomDestination(props: PropTypes) {
+  const { chain, handleCollapsibleOpenChange, lastStepToBlockchain } = props;
+  const { customDestination, setCustomDestination } = useQuoteStore();
+  const [open, setOpen] = useState(!!customDestination);
+  const { config } = useAppStore();
+  const blockchains = useAppStore().blockchains();
+  const blockchainDisplayName = getBlockchainDisplayNameFor(chain, blockchains);
+  const blockchainName = getBlockchainShortNameFor(chain, blockchains);
```

Comment:
You can simply use the chain from the properties. Using the short name causes a bug. You can pass a custom destination for blockchains like COSMOS or OSMOSIS and check if the custom destination is set.

Reply:
To avoid this issue, I used the wallet's own blockchain. Additionally, the value it requires matches up to tokens and blockchains parameters.